Assets/Scripts/Ability/AbilityManager.cs

```diff
@@ -87,21 +96,60 @@ private void OnDestroy()
         private void OnAbility1(InputAction.CallbackContext obj)
         {
             HandleAbilityBySlot((int)AbilitySlot.First);
+            _indicatorController.enabled = false;
+            _projector.enabled = false;
+            _slotActive = false;
         }
 
         private void OnAbility2(InputAction.CallbackContext obj)
         {
             HandleAbilityBySlot((int)AbilitySlot.Second);
+            _indicatorController.enabled = false;
+            _projector.enabled = false;
+            _slotActive = false;
         }
 
         private void OnAbility3(InputAction.CallbackContext obj)
         {
             HandleAbilityBySlot((int)AbilitySlot.Third);
+            _indicatorController.enabled = false;
+            _projector.enabled = false;
+            _slotActive = false;
+        }
+
+        private void OnAbilityStarted1(InputAction.CallbackContext obj)
+        {
+            if (_abilities.Count > 0 && _abilities[(int)AbilitySlot.First].CheckAvailability())
```

Comment:
👍 If read.
You can easily make this a function, and simply parse the enum (or index) in as a parameter, which would save you some typing.